### ISSUE TITLE
Can't run terragrunt apply

### ISSUE THREAD

**Issue by jamie3** [gruntwork-io/terragrunt#815]
I have a simple hello world terragrunt example and I am getting the following error:

```
$ tg apply
[terragrunt] [C:\dev\code\personal\terragrunt-hello-world\live\staging\app] 2019/08/02 15:35:43 Running command: terraform --version
[terragrunt] 2019/08/02 15:35:44 Reading Terragrunt config file at C:/dev/code/personal/terragrunt-hello-world/live/staging/app/terragrunt.hcl
[terragrunt] 2019/08/02 15:35:44 Downloading Terraform configurations from file://C:/dev/code/personal/terragrunt-hello-world/infrastructure into C:/dev/code/personal/terragrunt-hello-world/live/staging/app/.terragrunt-cache/Iuk57wqOPE_7PkKaFmUOii7vrSQ/jIMZ6meZTUEKM4-_y0agCmUjNao
[terragrunt] 2019/08/02 15:35:44 Hit multiple errors:
error downloading 'file://C:/dev/code/personal/terragrunt-hello-world/infrastructure': remove C:\dev\code\personal\terragrunt-hello-world\live\staging\app\.terragrunt-cache\Iuk57wqOPE_7PkKaFmUOii7vrSQ\jIMZ6meZTUEKM4-_y0agCmUjNao\app\myapp\main.tf: The system cannot find the file specified.
[terragrunt] 2019/08/02 15:35:44 Unable to determine underlying exit code, so Terragrunt will exit with error code 1
```

Steps to reproduct

```
git clone https://github.com/jamie3/terragrunt-hello-world
cd terragrunt-hello-world
cd live/staging/app
# First step works
terragrunt init   
# This fails
terragrunt apply
```

**Comment by yorinasub17** [comment]
You might be hitting the long paths issue with Windows. Can you try this workaround and see if that works? https://github.com/gruntwork-io/terragrunt/issues/581#issuecomment-460051767

**Comment by jamie3** [comment]
I tried the following:
```
export TERRAGRUNT_DOWNLOAD=/c/.terragrunt-cache
rm -rf .terraform-cache
terragrunt init
# Fail
terragrunt apply
```

FYI im running this inside VSCode bash shell

```
[terragrunt] [C:\dev\code\personal\terragrunt-hello-world\live\staging\app] 2019/08/02 15:58:55 Running command: terraform --version
[terragrunt] 2019/08/02 15:58:55 Reading Terragrunt config file at C:/dev/code/personal/terragrunt-hello-world/live/staging/app/terragrunt.hcl
[terragrunt] 2019/08/02 15:58:55 Downloading Terraform configurations from file://C:/dev/code/personal/terragrunt-hello-world/infrastructure into C:/.terragrunt-cache/Iuk57wqOPE_7PkKaFmUOii7vrSQ/jIMZ6meZTUEKM4-_y0agCmUjNao
[terragrunt] 2019/08/02 15:58:55 Hit multiple errors:
error downloading 'file://C:/dev/code/personal/terragrunt-hello-world/infrastructure': remove C:\.terragrunt-cache\Iuk57wqOPE_7PkKaFmUOii7vrSQ\jIMZ6meZTUEKM4-_y0agCmUjNao\app\myapp\main.tf: The system cannot find the file specified.        
[terragrunt] 2019/08/02 15:58:55 Unable to determine underlying exit code, so Terragrunt will exit with error code 1
```

**Comment by yorinasub17** [comment]
Which version of terragrunt are you using? I'm wondering if there is a bug from https://github.com/gruntwork-io/terragrunt/releases/tag/v0.19.13. Can you try https://github.com/gruntwork-io/terragrunt/releases/tag/v0.19.12?

**Comment by jamie3** [comment]
Sorry should have specified this terragrunt version v0.19.16

**Comment by jamie3** [comment]
That seemed to work. When I downgraded to 0.19.12 it worked

**Comment by yorinasub17** [comment]
FYI @brikis98 @ekini Looks like this is a bug from https://github.com/gruntwork-io/terragrunt/pull/774.

**Comment by ekini** [comment]
I think I know why it's happening. So I tried to clone the repo from scratch and wasn't able to reproduce the issue.

But this error might happen if:
1. You have `main.tf` in both `terraform_source` and the terragrunt module directory.
2. Run `terragrunt init` to copy the files. `main.tf` from terragrunt module will override the file copied first from `terraform_source`. When files are copied, they are added to 2 manifests, one for terraform source, and the other one from terragrunt module dir.
3. On a second run terragrunt will delete `main.tf` because it is in the manifest, and then try to delete it again, because it exists in the second manifest.
4. Previously, `os.RemoveAll` golang function was used without a comment, and it didn't make sense to use it to remove one file. Normally it's used to recursively delete files. I changed it to `os.Remove` as per the comment, and because it seemed logical. However, `os.RemoveAll` has a nice side effect: it doesn't return an error if the file doesn't exist already.

So I reverted it to using `RemoveAll` again, and added a comment explaining why this particular function is used. Hope it helps.

Thanks @jamie3 for the bug report!
